bundles/io/org.eclipse.smarthome.io.console/src/main/java/org/eclipse/smarthome/io/console/internal/extension/ItemConsoleCommandExtension.java

```diff
@@ -78,6 +99,42 @@ public void execute(String[] args, Console console) {
         }
     }
 
+    private void removeTag(String itemName, String tag, Console console) {
+        Item item = itemRegistry.get(itemName);
+        if (item instanceof GenericItem) {
+            GenericItem gItem = (GenericItem) item;
+            gItem.removeTag(tag);
+            Item oldItem = itemRegistry.update(gItem);
+            if (oldItem == null) {
+                console.println("Error: Cannot remove tag " + tag + " from item " + itemName
+                        + " because this item does not belong to a ManagedProvider");
+            } else {
+                console.println("Successfully removed tag " + tag + " from item " + itemName);
+            }
+        } else {
+            console.println(
+                    "Error: Cannot remove tag " + tag + " from item " + itemName + " because it is not a GenericItem");
+        }
+    }
+
+    private void addTag(String itemName, String tag, Console console) {
+        Item item = itemRegistry.get(itemName);
+        if (item instanceof GenericItem) {
+            GenericItem gItem = (GenericItem) item;
+            gItem.addTag(tag);
```

Comment:
The add and remove function are identical except the remove(d) / add(ed) string and the addTag / removeTag function.
As we are using Java 8 now, we can use method references, what do you think about moving the most part to a generic function that takes additional the method reference (`void ...(String tag)`) and two string (e.g. "add", "added" vs. "remove", "removed")?
We could leave it separated, but perhaps it could remove redundant code.